app/Models/User.php

```diff
@@ -102,6 +105,26 @@ public function subscriptions(): HasMany
         return $this->hasMany(Subscription::class);
     }
 
+    public function prospectSubscriptions(): MorphToMany
+    {
+        return $this->morphedByMany(
+            related: Prospect::class,
+            name: 'subscribable',
+            table: 'subscriptions'
+        )
+            ->using(Subscription::class);
```

Comment:
We don't necessarily need it at this point, but I'd recommend adding the `withPivot()` and `withTimestamps()` to these inverse relationships as well.